### PR TITLE
Maintain our own copy of pylint

### PR DESCRIPTION
**Pull request by ry**
depot_tools is removing pylint from its distribution so we must maintain
our own copy.

### CODE REVIEW

File: tools/lint.py [hunk 1]
```diff
@@ -24,5 +25,8 @@
     "./core/**/*.{ts,js}", "./tests/**/*.{ts,js}"
 ])
 
-run([sys.executable, "third_party/depot_tools/pylint.py"] +
```
Comment (ry) [inline tools/lint.py]: Note previously we called `third_party/depot_tools/pylint.py`

File: tools/complex_permissions_test.py [hunk 2]
```diff
@@ -31,16 +31,16 @@ def test_inside_project_dir(self):
...
+        assert PERMISSION_DENIED_PATTERN not in stderr
 
     def test_outside_test_dir(self):
         code, _stdout, stderr = self._run_deno([
             "--allow-" + self.test_type + "=" + os.path.join(
                 root_path, "tests")
         ], [self.test_type, "package.json"])
         assert code == 1
-        assert not PROMPT_PATTERN in stderr
+        assert PROMPT_PATTERN not in stderr
```
Comment (hayd) [inline tools/complex_permissions_test.py]: Aside.
we should probably be using `self.assertIn(PROMPT_PATTERN, stderr)`, which would give a nicer error message when this fails (it'd show stderr).

**Comment by piscisaureus** [comment]
Unreviewed pre-takeoff rubberstamp

File: tools/complex_permissions_test.py (hunk 2 above)
Comment (ry) [reply tools/complex_permissions_test.py]: Agreed. But let's do that in another PR.

**Pull request merged**